### PR TITLE
Fix remote name resolution in RepoUtil.java (fixes #1742)

### LINKED ISSUE #1742: bug: Cannot detect remote branch
### Cody Version

6.0.3

### IDE Information

IntelliJ IDEA 2024.1.1 (Ultimate Edition)
Build #IU-241.15989.150, built on April 29, 2024
Licensed to Indeed

Runtime version: 17.0.10+1-b1207.14 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.5
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 10
Metal Rendering is ON
Registry:
  ide.experimental.ui=true
  java.additional.external.annotations.root.url=file:///Users/mjiang/.gradle/indeed_ij_annotations
Non-Bundled Plugins:
  org.intellij.plugins.hcl (241.15989.49)
  com.jetbrains.space (241.15989.9)
  Pythonid (241.15989.194)
  intellij.jupyter (241.15989.194)
  com.intellij.lang.jsgraphql (241.14494.150)
  com.apollographql.ijplugin (4.0.0-beta.6)
  com.sourcegraph.jetbrains (6.0.3)
Kotlin: 241.15989.150-IJ

### Describe the bug

When I use `Copy Sourcegraph File Link` feature, the generated link always uses the fallback branch name (e.g. `main`), although I have confirmed that the corresponding changes have been pushed to remote.

I have already confirmed the authentication on my enterprise domain.

Did I miss anything in my configuration?
How can I debug that?

### Expected behavior

`Copy Sourcegraph File Link` could generate link with correct branch name instead of the fallback one. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mkondratek**
Fixes #1742 (https://linear.app/sourcegraph/issue/CODY-2077/bug-cannot-detect-remote-branch)

## Test plan
1. checkout a branch (not `main`)
2. right-click in the code, Sourcegraph, Copy Sourcegraph File Link

### Expected
The link refers the proper branch

### CODE REVIEW

File: src/main/java/com/sourcegraph/vcs/RepoUtil.java [hunk 1]
```diff
@@ -50,11 +50,10 @@ public static RepoInfo getRepoInfo(@NotNull Project project, @NotNull VirtualFil
...
 
-      // If the current branch doesn't exist on the remote or if the remote
-      // for the current branch doesn't correspond with the sourcegraph remote,
+      // If the current branch doesn't exist on the remote
       // use the default branch for the project.
       remoteBranchName = getRemoteBranchName(project, file);
-      if (remoteBranchName == null || !remoteUrl.contains(remoteBranchName)) {
+      if (remoteBranchName == null) {
         remoteBranchName = codyProjectSettings.getDefaultBranchName();
       }
```
Comment (mkondratek) [inline src/main/java/com/sourcegraph/vcs/RepoUtil.java:58]: @olafurpg, do you have any ideas what it was about?

Comment (olafurpg) [reply src/main/java/com/sourcegraph/vcs/RepoUtil.java:58]: Don't recall any specific details here

Comment (mkondratek) [reply src/main/java/com/sourcegraph/vcs/RepoUtil.java:58]: ok. anyway - it seems to work properly now

File: src/test/kotlin/com/sourcegraph/vcs/RepoUtilTest.kt [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+          /* name = */ GitRemote.ORIGIN,
+          /* urls = */ listOf("https://github.com/sourcegraph/jetbrains.git"),
+          /* pushUrls = */ emptyList(),
+          /* fetchRefSpecs = */ emptyList(),
+          /* pushRefSpecs = */ emptyList())
+
+  private val gitLocalBranch = GitLocalBranch("mkondratek/great-new-feature")
+  private val gitRemoteBranch = GitStandardRemoteBranch(gitRemote, "mkondratek/great-new-feature")
+
+  override fun dispose() {
```
Comment (dominiccooney) [inline src/test/kotlin/com/sourcegraph/vcs/RepoUtilTest.kt]: This is fine, but why not use Mockito to mock this, too? You won't have to write out all these TODOs... which aren't really TODOs...

**Review by dominiccooney — APPROVED** [review]
Approving to unblock but consider using `mock` for succinctness. The TODOs are not really TODOs.

File: src/test/kotlin/com/sourcegraph/vcs/RepoUtilTest.kt (hunk 2 above)
Comment (mkondratek) [reply src/test/kotlin/com/sourcegraph/vcs/RepoUtilTest.kt]: good point, thanks

**Pull request merged**